node/src/chain/consensus.rs

```diff
@@ -158,22 +159,37 @@ impl<DB: database::DB> dusk_consensus::commons::Database for CandidateDB<DB> {
 }
 
 /// Implements Executor trait to mock Contract Storage calls.
-pub struct Executor<DB: database::DB> {
+pub struct Executor<DB: database::DB, VM: vm::VMExecution> {
     db: Arc<RwLock<DB>>,
+    vm: Arc<RwLock<VM>>,
 }
 
-impl<DB: database::DB> Executor<DB> {
-    fn new(db: &Arc<RwLock<DB>>) -> Self {
-        Executor { db: db.clone() }
+impl<DB: database::DB, VM: vm::VMExecution> Executor<DB, VM> {
+    fn new(db: &Arc<RwLock<DB>>, vm: &Arc<RwLock<VM>>) -> Self {
+        Executor {
+            db: db.clone(),
+            vm: vm.clone(),
+        }
     }
 }
 
-impl<DB: database::DB> Operations for Executor<DB> {
+impl<DB: database::DB, VM: vm::VMExecution> Operations for Executor<DB, VM> {
     fn verify_state_transition(
         &self,
-        _params: CallParams,
+        params: CallParams,
     ) -> Result<StateRoot, dusk_consensus::contract_state::Error> {
         tracing::info!("verifying state");
+
+        let vm = self.vm.try_read().map_err(|e| {
+            tracing::error!("failed to try_read vm: {}", e);
+            Error::Failed
+        })?;
```

Comment:
Agree with you.
 There is not reason to see `try_read` failing. However, I'd also prefer to use `await` but at the time of implementing this I wanted to avoid bumping again consensus crate due to making `Operations` `async_trait`.

NB: As you know, updating/bumping the three crates (node, node-data, consensus) became pretty cumbersome and we should find a way to get rid of this obstacle.